### ISSUE TITLE
S3 add bucket and path as meta information

### ISSUE THREAD

**Issue by jordansissel** [logstash-plugins/logstash-input-s3#39]
(This issue was originally filed by @pvanisacker at https://github.com/elastic/logstash/issues/1840)

---

Currently when S3 file get imported the name of the file and the bucket is not stored.
It would be nice to have that as a field as well.

**Comment by pliljenberg** [comment]
Would love to get the PR merged and a new version released.

**Comment by jeffxor** [comment]
Not sure what this pull request is waiting on. It is failing because of not signing CLA but it looks like other work from this submitter has already been approved. Looks like he works for elastic?

**Comment by srang** [comment]
+1 could really use this functionality

**Comment by DrValani** [comment]
+1. Really need this functionality. 

This looks really similar to PR #76.

**Comment by lattwood** [comment]
Hey @ph, we could really use this plugin too, what's holding the merge up?

**Comment by alfwed** [comment]
+1 This plugin is useless without this functionality

**Comment by bobbrez** [comment]
+1 on this, it seems like this functionality exists in a few PRs... Is there anything that we can do to move this forward @ph, @purbon, or @jsvd

**Comment by ivanschwarz** [comment]
+1 on this. Is there a way to get this moving forward? I can see both the issue and a solution already exist for quite a while.

**Comment by jordansissel** [comment]
@ivanschwarz Good question!

If you believe #76 to be the right solution, the way anyone can help it move forward is to help with review of the proposal -- Here are a few questions I carry when reviewing code:

* Does the PR solve the problem?
* Does it break backwards compatibility?
* Does the new functionality have documentation?
* Does it have tests that cover the new functionality?
* Do the tests pass?
* Is the code reasonably clear?
* Does the new functionality work? (manual testing). The way you expect?

The list above is incomplete, but having anyone work on answering any of them is a good step towards getting the patch merged.

**Comment by ph** [comment]
The key will be in 3.1.3
